Tests/Form/FormMapperTest.php

```diff
@@ -429,6 +429,20 @@ public function testKeys()
         $this->assertSame(array('foo', 'baz'), $this->formMapper->keys());
     }
 
+    public function testFieldNameIsSanitized()
+    {
+        $this->contractor->expects($this->any())
+            ->method('getDefaultOptions')
+            ->will($this->returnValue(array()));
+
+        $this->formMapper
+            ->add('fo.o', 'bar')
+            ->add('ba__z', 'foobaz')
+        ;
+
+        $this->assertSame(array('fo__o', 'ba____z'), $this->formMapper->keys());
```

Comment:
@jlamur if we expect to have `ba__z` when `ba.z` is added we can't expect to have `ba__z` when `ba__z`, the sanitize method has to be reversible.